keadm/app/cmd/version.go

```diff
@@ -0,0 +1,93 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	yaml "gopkg.in/yaml.v2"
+	"k8s.io/klog"
+
+	"github.com/kubeedge/kubeedge/pkg/version"
+)
+
+const (
+	// DefaultErrorExitCode defines exit the code for failed action generally
+	DefaultErrorExitCode = 1
+)
+
+func NewCmdVersion(out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "version",
+		Short: "Print the version of kubeadm",
+		Run: func(cmd *cobra.Command, args []string) {
+			err := RunVersion(out, cmd)
+			CheckErr(err, fatal)
+		},
+	}
+	cmd.Flags().StringP("output", "o", "", "Output format; available options are 'yaml', 'json' and 'short'")
+	return cmd
+}
+
+// RunVersion provides the version information of kubeadm in format depending on arguments
+// specified in cobra.Command.
+func RunVersion(out io.Writer, cmd *cobra.Command) error {
+	v := version.Get()
+
+	const flag = "output"
+	of, err := cmd.Flags().GetString(flag)
+	if err != nil {
+		klog.Fatalf("error accessing flag %s for command %s: %v", flag, cmd.Name(), err)
+	}
+
+	switch of {
+	case "":
+		fmt.Fprintf(out, "version: %#v\n", v)
+	case "short":
+		fmt.Fprintf(out, "%s\n", v)
+	case "yaml":
+		y, err := yaml.Marshal(&v)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintln(out, string(y))
+	case "json":
+		y, err := json.MarshalIndent(&v, "", "  ")
+		if err != nil {
+			return err
+		}
+		fmt.Fprintln(out, string(y))
+	default:
+		return errors.Errorf("invalid output format: %s", of)
```

Comment:
can we use fmt.Errorf instead of errors.Errorf? It would decrease the count of dependencies

Reply:
we refer to [kubernetes code ](https://github.com/kubernetes/kubernetes/blob/4495d09282265964ae217ab9976c4c4ff5945b7b/cmd/kubeadm/app/cmd/version.go#L86)